extendeddatacrossword_test.go

```diff
@@ -175,6 +163,23 @@ func BenchmarkRepair(b *testing.B) {
 				),
 				func(b *testing.B) {
 					for n := 0; n < b.N; n++ {
+						b.StopTimer()
+
+						flattened := eds.flattened()
+						// Randomly remove 1/2 of the shares of each row
+						for r := 0; r < extendedDataWidth; r++ {
+							for c := 0; c < originalDataWidth; {
+								ind := rand.Intn(extendedDataWidth)
+								if flattened[r*extendedDataWidth+ind] == nil {
+									continue
+								}
+								flattened[r*extendedDataWidth+ind] = nil
+								c++
+							}
+						}
+
+						b.StartTimer()
+
 						_, err := RepairExtendedDataSquare(
```

Comment:
Shouldn't this actually set the eds back to fully repaired though with each iteration? I see that the benchmarks work  without it but I'm not sure I understand why tbh.

Reply:
Actually, NVM. It's only the `flattened` slice that gets modified on each iteration and not the EDS...